src/ansys/fluent/core/filereader/case.py

```diff
@@ -99,7 +100,11 @@ def __call__(self, name: str = ""):
         if not name:
             error_name = self._path[:-1] if self._path else self._path
             raise RuntimeError(f"Invalid variable {error_name}")
-        return self._variables[name]
+        try:
+            return self._variables[name]
+        except KeyError:
+            raise RuntimeError(allowed_name_error_message(
```

Comment:
I think we can throw [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) here for out-of-bound arguments.